packages/uploadthing/test/adapters.test.ts

```diff
@@ -0,0 +1,310 @@
+    expect(fetchMock).toHaveBeenCalledTimes(1);
+    expect(fetchMock).toHaveBeenCalledWith(
+      "https://uploadthing.com/api/prepareUpload",
+      {
+        body: '{"files":[{"name":"foo.txt","size":48}],"routeConfig":{"blob":{"maxFileSize":"8MB","maxFileCount":1,"contentDisposition":"inline"}},"metadata":{},"callbackUrl":"https://not-used.com/","callbackSlug":"middleware"}',
+        headers: {
+          "Content-Type": "application/json",
+          "x-uploadthing-api-key": "sk_live_test",
+          "x-uploadthing-be-adapter": "nextjs-app",
+          "x-uploadthing-fe-package": "vitest",
+          "x-uploadthing-version": expect.stringMatching(/\d+\.\d+\.\d+/),
+        },
+        method: "POST",
+      },
+    );
+  });
+});
+
+describe("adapters:next-legacy", async () => {
+  const { createUploadthing, createRouteHandler } = await import(
+    "../src/next-legacy"
+  );
+  const f = createUploadthing();
+
+  const router = {
+    middleware: f({ blob: {} })
+      .middleware((opts) => {
+        middlewareMock(opts);
+        expectTypeOf<{
+          event: undefined;
+          req: NextApiRequest;
+          res: NextApiResponse;
+        }>(opts);
+        return {};
+      })
+      .onUploadComplete(uploadCompleteMock),
+  };
+
+  const handler = createRouteHandler({
+    router,
+    config: {
+      uploadthingSecret: "sk_live_test",
+      fetch: mockExternalRequests,
+    },
+  });
+
+  function mockReq(opts: {
+    query: Record<string, any>;
+    headers?: Record<string, string>;
+    method?: string;
+    body?: unknown;
+  }) {
+    const req = new EventEmitter() as any;
+
+    req.url = "/?" + new URLSearchParams(opts.query).toString();
+    req.method = opts.method;
+    req.query = opts.query;
+    req.headers = {
+      "x-forwarded-host": "not-used.com",
+      "x-forwarded-proto": "https",
+      ...opts.headers,
+    };
+    req.body = opts.body;
+
+    const socket = {
+      destroy: vi.fn(),
+    };
+    req.socket = socket;
+
+    return { req, socket };
+  }
+  function mockRes() {
+    const res = new EventEmitter() as any;
+
+    const json = vi.fn(() => res);
+    const setHeader = vi.fn(() => res);
+    const status = vi.fn(() => res);
+    res.json = json;
+    res.setHeader = setHeader;
+    res.status = status;
+
+    return { res, json, setHeader, status };
+  }
+
+  it("gets NextApiRequest and NextApiResponse in middleware args", async () => {
+    const { req } = mockReq({
+      query: { slug: "middleware", actionType: "upload" },
+      body: {
+        files: [{ name: "foo.txt", size: 48 }],
+      },
+      method: "POST",
+      headers: baseHeaders,
+    });
+    const { res, status } = mockRes();
+
+    await handler(req, res);
+    expect(status).toHaveBeenCalledWith(200);
+
+    expect(middlewareMock.mock.calls[0][0].event).toBeUndefined();
+    expect(middlewareMock.mock.calls[0][0].req).toHaveProperty("query");
+    expect(middlewareMock.mock.calls[0][0].res).toHaveProperty("json");
```

Comment:
`NextApiRequest` and `NextApiResponse` aren't classes so can't do `toBeInstanceOf`